### PR TITLE
Add inspection v2 (always encrypted) option.

### PR DESCRIPTION
**Pull request by klauspost**
Depends on: https://github.com/minio/minio/pull/15474 for functionality (which depends on this)

Blocks:  https://github.com/minio/minio/pull/15474

Keep v1 if non-encrypted and decrypt it.

Otherwise if encrypted, use a fixed public key.

Will not work until minio is upgraded with as well.

### CODE REVIEW

**Comment by klauspost** [comment]
Still pending [mc 4185](https://github.com/minio/mc/pull/4185), but short of `go.mod` updates it is ready for review.

**Comment by harshavardhana** [comment]
@klauspost PTAL at the tests

```
=== RUN   TestInspect
=== RUN   TestInspect/Test_Invalid_Path
=== RUN   TestInspect/Test_Invalid_characters_in_Path
=== RUN   TestInspect/Test_valid_bucket
=== CONT  TestInspect
    inspect_test.go:99: 
        	Error Trace:	/home/runner/work/console/console/integration/inspect_test.go:99
        	Error:      	Not equal: 
        	            	expected: 200
        	            	actual  : 500
        	Test:       	TestInspect
=== RUN   TestInspect/Test_Empty_Path
--- FAIL: TestInspect (0.03s)
    --- PASS: TestInspect/Test_Invalid_Path (0.01s)
    --- PASS: TestInspect/Test_Invalid_characters_in_Path (0.01s)
    --- PASS: TestInspect/Test_valid_bucket (0.01s)
    --- PASS: TestInspect/Test_Empty_Path (0.01s)
=== RUN   TestLoginStrategy
```

**Comment by klauspost** [comment]
@harshavardhana It is a cyclic dependency. Minio depends on this and this depends on minio

**Comment by harshavardhana** [comment]
> @harshavardhana It is a cyclic dependency. Minio depends on this and this depends on minio

Then can we skip it? @klauspost as a non-failure?

**Review by harshavardhana — DISMISSED** [review]

File: go.mod [hunk 1]
```diff
@@ -192,3 +192,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/minio/mc => github.com/klauspost/mc v0.0.0-20221014120903-10af4c8890f4
```
Comment (bexsoft) [inline go.mod:196]: Please remove this replace

File: restapi/admin_inspect.go [hunk 2]
```diff
@@ -37,28 +35,40 @@ import (
...
-	cfg.Volume = params.Volume
+	cfg := madmin.InspectOptions{
+		File:   params.File,
+		Volume: params.Volume,
+	}
+
+	// TODO: Remove encryption option and always encrypt.
+	// Maybe also add public key field.
+	if params.Encrypt != nil && *params.Encrypt {
+		cfg.PublicKey, _ = base64.StdEncoding.DecodeString("MIIBCgKCAQEAs/128UFS9A8YSJY1XqYKt06dLVQQCGDee69T+0Tip/1jGAB4z0/3QMpH0MiS8Wjs4BRWV51qvkfAHzwwdU7y6jxU05ctb/H/WzRj3FYdhhHKdzear9TLJftlTs+xwj2XaADjbLXCV1jGLS889A7f7z5DgABlVZMQd9BjVAR8ED3xRJ2/ZCNuQVJ+A8r7TYPGMY3wWvhhPgPk3Lx4WDZxDiDNlFs4GQSaESSsiVTb9vyGe/94CsCTM6Cw9QG6ifHKCa/rFszPYdKCabAfHcS3eTr0GM+TThSsxO7KfuscbmLJkfQev1srfL2Ii2RbnysqIJVWKEwdW05ID8ryPkuTuwIDAQAB")
```
Comment (bexsoft) [inline restapi/admin_inspect.go:70]: Do we need this Public Key here?

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: restapi/admin_inspect.go (hunk 2 above)
Comment (klauspost) [reply restapi/admin_inspect.go:70]: Yes, we must always encrypt.

Comment (klauspost) [reply restapi/admin_inspect.go:70]: As noted, the UI could be made to ask for a public key.

The un-encrypted option should be removed from the UI, but that is beyond my capabilities.

**Review by harshavardhana — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]
Cannot generate inspect files with encryption set as true:
Screenshot 2022-11-02 at 10 31 57

This seems related to the missing dependency from minio, will approve this to unblock the rest of required PRs

**Comment by klauspost** [comment]
@bexsoft Yes, the encryption option requires the minio PR to be running to work (and the PR above requires this console version).

**Pull request merged**